drivers/mfd/stm32_timers.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2019 Max van Kessel
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_MFD_STM32_TIMERS_H_
+#define ZEPHYR_DRIVERS_MFD_STM32_TIMERS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct stm32_timers {
+    TIM_TypeDef *tim;       /**< Base address */
+    struct device *clock;   /**< Clock device */
+};
+
+/**
+ * @typedef stm32_timers_enable_t
+ *
+ * @brief Enable stm32 timer device
+ * @param dev	The device to enable
+ */
+typedef void (*stm32_timers_enable_t)(struct device *dev);
+
+/**
+ * @typedef stm32_timers_disable_t
+ *
+ * @brief Disable stm32 timer device
+ * @param dev	The device to disable
+ */
+typedef void (*stm32_timers_disable_t)(struct device *dev);
+
+/**
+ * @typedef stm32_timers_get_cycles_per_sec_t
+ *
+ * @brief Get the clock rate (cycles per second)
+ * @param dev		Pointer to timer device structure
+ * @param cycles	Pointer to the memory to store clock rate
+* 			(cycles per second)
+ * @return 0 for success
+ * @return negative errno code
+ */
+typedef int (*stm32_timers_get_cycles_per_sec_t)(struct device *dev,
+							u64_t * cycles);
+
+struct stm32_timers_api {
+	stm32_timers_enable_t enable;
+	stm32_timers_disable_t disable;
+	stm32_timers_get_cycles_per_sec_t get_cycles_per_sec;
+};
+
+
+static inline void stm32_timers_enable(struct device *dev)
```

Comment:
Are these available to user space as well?
If yes, you should add __syscall function similarly to what is done in other API.

Reply:
On creation my intention was, that this driver can only be called by system drivers. 
But I use it within a user-space driver, so will change this.

Reply:
Makes sense, fine.